torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -628,6 +647,13 @@ void ProcessGroupNCCL::workCleanupLoop() {
         ++it;
       }
     }
+
+    if (workList_.empty()) {
+      // Notify the main thread if it is blocked in the shutdown sequence,
+      // waiting for the work vector to become empty.
+      lock.unlock();
+      workListCV_.notify_one();
+    }
```

Comment:
Seems destructor is not waiting for workList to be empty(). Wondering if somewhere else is waiting for this? or we can just remove?